lib/ElementHandle.js

```diff
@@ -137,24 +140,67 @@ class ElementHandle extends JSHandle {
     return {x, y, width, height};
   }
 
+  /**
+   * @return {!Promise<?{x: number, y: number, width: number, height: number}>}
+   */
+  async _assertBoundingBox() {
+    const boundingBox = await this.boundingBox();
+    if (boundingBox)
+      return boundingBox;
+
+    const noElementError = await this.executionContext().evaluate(element => {
```

Comment:
I'd save a roundtrip and throw a generic error: `Node is either not visible or not an HTMLElement`.

Reply:
Fixed. I agree that we're not losing much useful information in this case.